plugins/slo/pkg/slo/ir.go

```diff
@@ -0,0 +1,216 @@
+package slo
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+	"regexp"
+	"strings"
+	"time"
+
+	openslov1 "github.com/alexandreLamarre/oslo/pkg/manifest/v1"
+	"github.com/alexandreLamarre/sloth/core/app/generate"
+	"github.com/alexandreLamarre/sloth/core/info"
+	"github.com/alexandreLamarre/sloth/core/prometheus"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	specTypeV1AlphaRegexKind       = regexp.MustCompile(`(?m)^kind: +['"]?SLO['"]? *$`)
+	specTypeV1AlphaRegexAPIVersion = regexp.MustCompile(`(?m)^apiVersion: +['"]?openslo\/v1alpha['"]? *$`)
+)
+
+var errorRatioRawQueryTpl = template.Must(template.New("").Parse(`
+  1 - (
+    (
+      {{ .good }}
+    )
+    /
+    (
+      {{ .total }}
+    )
+  )
+`))
```

Comment:
You should strip the leading newline from this string by appending `[1:]` after the closing `